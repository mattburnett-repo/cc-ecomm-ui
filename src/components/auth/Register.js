export default function Register() {
    return (
        <div>
            <div>
                <button aria-label="google-auth">Log in with Google</button>
                <button aria-label="github-auth">Log in with Google</button>
            </div>
            <div>
                <label htmlFor="username">Username:</label>
                <input id="username" aria-label="username" />
            </div>
            <div>
                <label htmlFor="email">Email:</label>
                <input id="email" type="email" aria-label="email" />
            </div>
            <div>
                <label htmlFor="password">Password:</label>
                <input id="password" type="password" aria-label="password" />
            </div>
            <div>
                <button aria-label="register">Register</button>
            </div>
            <div>
                <a href="/login" aria-label="login">Login</a>
            </div>
        </div>
    )
}