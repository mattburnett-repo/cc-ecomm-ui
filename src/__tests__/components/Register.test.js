// 20211217: Wireframe https://wireframe.cc/6MXHyR

import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

import Register from '../../components/auth/Register';

describe('<Register /> component tests', () => {
    beforeEach(() => render(<Register />));

    it('should display a google oauth button', () => {
        screen.getByRole('button', { name: /google-auth/i});
    });
    
    it('should display a github oauth button', () => {
        screen.getByRole('button', { name: /github-auth/i});
    });
    it('should display a username input', () => {
        screen.getByRole('textbox', { name: /username/i});
    });
    it('should display an email input', () => {
        screen.getByRole('textbox', { name: /email/i});
    });
    it('should display a password input', () => {
        screen.getByLabelText('Password:');
    });
    it('should display a Register button', () => {
        screen.getByRole('button', { name: /Register/i});
    });
    it('should have a Log in link', () => {
        screen.getByRole('link', { name: /login/i});
    })
}); // end component

// features?

describe('<Register /> renders a snapshot', () => {
    it('renders a snapshot', () => {
        const tree = renderer.create(<Register />).toJSON();
        expect(tree).toMatchSnapshot();
    });
})

describe('<Register /> unit tests', () => {
    beforeEach(() => render(<Register />));

    it('clicks the Google auth button', () => {
        const theVal = screen.getByRole('button', {name: /google-auth/i});
        fireEvent.click(theVal);

        // TODO: mock an onSubmit function and test that also
    });
    it('clicks the GitHub auth button', () => {
        const theVal = screen.getByRole('button', {name: /github-auth/i});
        fireEvent.click(theVal);

        // TODO: mock an onSubmit function and test that also
    });
    it('accept a value for username', ()=> {
        const theElement = screen.getByRole('textbox', {name: /username/i});
        userEvent.type(theElement, 'asdf');
        
        expect(theElement).toHaveValue('asdf');
    });
    it('accept a value for password', ()=> {
        const theElement = screen.getByLabelText('Password:');
        userEvent.type(theElement, 'asdf');
        
        expect(theElement).toHaveValue('asdf');
    });
    it('clicks the Register button', () => {
        const theVal = screen.getByRole('button', {name: /Register/i});
        fireEvent.click(theVal);

        // TODO: mock an onSubmit function and test that also
    });
    it('clicks the Login link', () => {
        const theVal = screen.getByRole('link', {name: /login/i});
        fireEvent.click(theVal);
    });
}); // end unit
