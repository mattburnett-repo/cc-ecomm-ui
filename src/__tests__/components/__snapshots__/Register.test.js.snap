// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> renders a snapshot renders a snapshot 1`] = `
<div>
  <div>
    <button
      aria-label="google-auth"
    >
      Log in with Google
    </button>
    <button
      aria-label="github-auth"
    >
      Log in with Google
    </button>
  </div>
  <div>
    <label
      htmlFor="username"
    >
      Username:
    </label>
    <input
      aria-label="username"
      id="username"
    />
  </div>
  <div>
    <label
      htmlFor="email"
    >
      Email:
    </label>
    <input
      aria-label="email"
      id="email"
      type="email"
    />
  </div>
  <div>
    <label
      htmlFor="password"
    >
      Password:
    </label>
    <input
      aria-label="password"
      id="password"
      type="password"
    />
  </div>
  <div>
    <button
      aria-label="register"
    >
      Register
    </button>
  </div>
  <div>
    <a
      aria-label="login"
      href="/login"
    >
      Login
    </a>
  </div>
</div>
`;
